astropy/nddata/nduncertainty.py

```diff
@@ -3,17 +3,71 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import abc
-
 import numpy as np
+from abc import ABCMeta, abstractproperty, abstractmethod
+from copy import deepcopy
 
-from ..utils.compat import ignored
-from ..units import Quantity
+# from ..utils.compat import ignored
+from .. import log
+from ..units import Unit, Quantity
 from ..extern import six
 
 __all__ = ['MissingDataAssociationException',
            'IncompatibleUncertaintiesException', 'NDUncertainty',
-           'StdDevUncertainty']
+           'StdDevUncertainty', 'UnknownUncertainty']
+
+try:
+    from ..utils import format_doc
+except ImportError:
+    # TODO: Delete this and rebase if #4242 is merged.
+    def format_doc(docstring, *args, **kwargs):
+        def set_docstring(func):
+            if not isinstance(docstring, six.string_types):
+                doc = docstring.__doc__
+            elif docstring != 'self':
+                doc = docstring
+            else:
+                doc = func.__doc__
+                func.__doc__ = None
+            if not doc:
+                raise ValueError
+            kwargs['original_doc'] = func.__doc__ or ''
+            func.__doc__ = doc.format(*args, **kwargs)
+            return func
+        return set_docstring
+
+# Make a placeholder for the different uncertainty propagation methods.
+_propagate_doc = """
+Propagate uncertainties for {operation}.
+
+Parameters
+----------
+other_uncert : `{instance}`
+    The data for the uncertainty of b in a {operator} b
+result_data : `~numpy.ndarray` instance or `~astropy.units.Quantity`
+    The data array that is the result of the {operation}.
+correlation: `Number` or `~numpy.ndarray`
+    Array or scalar representing the correlation. If the subclass does not
+    support correlated uncertainties this will be replaced by 0 (uncorrelated).
```

Comment:
Haven't read the code below yet, so ignore if the code addresses this, but it seems dangerous to silently change correlated uncertainties to uncorrelated ones.